src/nccl_ofi_mr.c

```diff
@@ -147,16 +147,15 @@ void *nccl_ofi_mr_cache_lookup_entry(nccl_ofi_mr_cache_t *cache,
 }
 
 int nccl_ofi_mr_cache_insert_entry(nccl_ofi_mr_cache_t *cache,
-				   void *addr,
-				   size_t size,
-				   void *handle)
+								   const nccl_ofi_mr_input_t *in,
```

Comment:
I'm not sure I see the reason to change this function either.  The cache is only concerned about virtual address ranges, and gets back an MR that covers that.  Since we're not doing MR 0-based addressing, we don't actually care about anything other than the VA range of the MR, and shouldn't care at all about whether the MR is a dmabuf mr or not.

Reply:
Then this isn't a useful change.  Remove it.

Reply:
> Since we're not doing MR 0-based addressing,

I think we should plan to support this some day, toward supporting non-EFA providers? The TCP provider seems to use 0-based MR addressing.

Reply:
This would be a fairly significant amount of work; we're never going to write that code.  Maybe someone else contributes it, but it isn't something we should plan on.

Reply:
> Then this isn't a useful change.

Not following. You're saying that the cache shouldn't care about whether it's a dmabuf MR or not. I agree -- but if we continue to accept a base address and length on this interface, then the caller still does have to worry about it.

After this change, neither callers nor the cache implementation itself needs to concern itself with whether it's a dmabuf MR or not. Callers forward a pointer to a generic `nccl_ofi_mr_input` that can contain either type of mr, the cache implementation calls `nccl_ofi_mr_input_base` and `nccl_ofi_mr_input_len` which will return the right base and len for either type of mr, and it's really difficult to imagine how someone could pass the wrong base address or length into their cache query unintentionally. 

Objectively, that's an advantage of doing it this way. What am I missing here? What's so complicated about a tagged union?